From 49fb2ee3887979e3e86708cf8eff7d2374a90d80 Mon Sep 17 00:00:00 2001
From: Wenli Looi <wlooi@ucalgary.ca>
Date: Tue, 11 Apr 2023 05:21:04 +0000
Subject: [PATCH v3 04/11] Revert "ath9k_hw: fall back to OTP ROM when platform
 data has no valid eeprom data"

This reverts commit 01967360a20598f95947a86af26c930ab1b50205.

QCN5502 is a 4-stream device, so 0xff is the normal txrxMask. Users
currently relying on this silent fallback will need to stop providing
invalid EEPROM data to the driver.

Signed-off-by: Wenli Looi <wlooi@ucalgary.ca>
---
 drivers/net/wireless/ath/ath9k/ar9003_eeprom.c | 15 ++-------------
 1 file changed, 2 insertions(+), 13 deletions(-)

diff --git a/drivers/net/wireless/ath/ath9k/ar9003_eeprom.c b/drivers/net/wireless/ath/ath9k/ar9003_eeprom.c
index 3dab290d8a..0f8af8c958 100644
--- a/drivers/net/wireless/ath/ath9k/ar9003_eeprom.c
+++ b/drivers/net/wireless/ath/ath9k/ar9003_eeprom.c
@@ -3321,21 +3321,10 @@ static int ar9300_eeprom_restore_internal(struct ath_hw *ah,
 	int it;
 	u16 checksum, mchecksum;
 	struct ath_common *common = ath9k_hw_common(ah);
-	struct ar9300_eeprom *eep;
 	eeprom_read_op read;
 
-	if (ath9k_hw_use_flash(ah)) {
-		u8 txrx;
-
-		if (ar9300_eeprom_restore_flash(ah, mptr, mdata_size))
-			return -EIO;
-
-		/* check if eeprom contains valid data */
-		eep = (struct ar9300_eeprom *) mptr;
-		txrx = eep->baseEepHeader.txrxMask;
-		if (txrx != 0 && txrx != 0xff)
-			return 0;
-	}
+	if (ath9k_hw_use_flash(ah))
+		return ar9300_eeprom_restore_flash(ah, mptr, mdata_size);
 
 	word = kzalloc(2048, GFP_KERNEL);
 	if (!word)
-- 
2.34.1

